lbz/events/api.py

```diff
@@ -0,0 +1,75 @@
+import json
+from os import getenv
+from typing import TYPE_CHECKING, List
+
+from lbz.aws_boto3 import client
+from lbz.misc import Singleton
+
+if TYPE_CHECKING:
+    from mypy_boto3_events.type_defs import PutEventsRequestEntryTypeDef
+else:
+    PutEventsRequestEntryTypeDef = dict
+
+
+class BaseEvent:
+    type: str
+
+    def __init__(self, raw_data: dict) -> None:
+        self.raw_data = raw_data
+        self.data: str = self.serialize(raw_data)
+
+    @staticmethod
+    def serialize(raw_data: dict) -> str:
+        return json.dumps(raw_data, default=str)
+
+
+class EventAPI(metaclass=Singleton):
+    def __init__(self) -> None:
+        self._source = getenv("AWS_LAMBDA_FUNCTION_NAME") or "lbz-event-api"
+        self._resources: List[str] = []
+        self._pending_events: List[BaseEvent] = []
+        self._sent_events: List[BaseEvent] = []
+        self._bus_name = getenv("EVENT_BRIDGE_BUS_NAME", f"{self._source}-event-bus")
+
+    def __repr__(self) -> str:
+        return f"<EventAPI bus: {self._bus_name} pending_events_count: {len(self)}>"
+
+    def __len__(self) -> int:
+        return len(self._pending_events)
+
+    def set_source(self, source: str) -> None:
+        self._source = source
+
+    def set_resources(self, resources: List[str]) -> None:
+        self._resources = resources
+
+    def register(self, new_event: BaseEvent) -> None:
+        self._pending_events.append(new_event)
+
+    def get_all_pending_events(self) -> List[BaseEvent]:
+        return self._pending_events
+
+    def get_all_sent_events(self) -> List[BaseEvent]:
+        return self._sent_events
+
+    def send(self) -> None:
+        if self._pending_events:
+            entries = [self._create_eb_entry(event) for event in self._pending_events]
+            client.eventbridge.put_events(Entries=entries)
+            self._mark_sent()
+
+    def _create_eb_entry(self, new_event: BaseEvent) -> PutEventsRequestEntryTypeDef:
+        """
+        boto3.amazonaws.com/v1/documentation/api/latest/reference/services/events.html#EventBridge.Client.put_events
```

Comment:
But for me, it can be removed at all as everything is described by `boto3-stubs` - this is better to rely on it instead of putting links to documentation every time we use something from `boto3` 🙂